server-spi/src/main/java/com/flipkart/varadhi/spi/authn/Authenticator.java

```diff
@@ -0,0 +1,13 @@
+package com.flipkart.varadhi.spi.authn;
+
+
+import com.flipkart.varadhi.entities.auth.UserContext;
+import com.flipkart.varadhi.spi.RequestContext;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.vertx.core.Future;
+
+public interface Authenticator {
```

Comment:
NIT: noticing that we have authorizationProvider but this is Authenticator. so may be we can call the other Authorizer? :) or this one as AuthenticationProvider?

Reply:
AuthenticationProvider is an interface already provided by vertx, and it was getting confusing during discussions. 
we can rename AuthorisationProvider to Authorizer, since it is also not providing anything.